obstacledetection/src/main/java/com/caneseeproject/obstacledetection/ObstacleDetectionInterface.kt

```diff
@@ -25,23 +22,37 @@ sealed class ODReading : SensorReading {
     //TODO any other type of reading
 }
 
-sealed class ODInput : SensorInput {
+sealed class ODControl : SensorControl {
 
     //Represented with a '0' -> 0_percentage
-    class RangeControl(val percentage: Int) : ODInput()
+    class RangeControl(val percentage: Int) : ODControl()
 
     //TODO any other type of input
 
 }
 
-interface ObstacleDetector : Sensor<ODInput, ODReading> {
+interface ObstacleDetector : Sensor<ODReading, ODControl>{
+
+    /**
+     * Connecting with this cane
+     */
+    fun activate()
 
     /**
      * Provides flow of high level sensor readings
      */
-    fun detectObstacles(): Flow<ODReading.ObstacleDistance>
+    fun getCaneData(): Flow<ODReading>
```

Comment:
kotlin in general hates "getters" and "setters", you should too..

Reply:
ok